### ISSUE TITLE
Multires zoom changes on full screen.

### ISSUE THREAD

**Issue by MmmDee** [mpetroff/pannellum#187]
I'm trying out `multires` for the first time. Source of panorama was a Google street view photosphere, 338 tiles, each 512x512 (26 columns x 13 rows) which were then stitched together using graphicsmagick making `muni.jpg` and that appears very "normal" when NOT using multires.  Then I used` generate.py` thusly, `python generate.py -o muni -s 512 muni.jpg`. I used the API from the example on your website, changing `basePath`, `maxLevel` and `cubeResolution` (per the generated `config.json` file).  All is well so far. 

I then bring up the webpage at 600x400 and all works as expected: rotate and zoom. Then I zoom-in to the max and while there, I then click on the full screen icon. That works. But as soon as I then touch the zoom-in/out icons or scroll the mouse wheel, the full screen (remains full screen but) jumps "out" a level and won't return when the zoom-in icon is again clicked.  I can "rotate" the image without any consequence.

This is using the latest commit [2ca29f7b](https://github.com/mpetroff/pannellum/commit/2ca29f7b7ef1ad0afd452fd8dbbf597caa1aade8).

The same behavior occurs on an image from the Theta.

Is this expected?

Perhaps unrelated, I note many "tile" artifacts when using the latest commit vs the release versions of pannellum. With the release, I saw no "tile" artifacts (a tile artifact is a black square, triangle, other polygon) appearing randomly on the screen for a few  milliseconds to a second or more before the artifact is replaced with an image fragment (sometimes the artifact is never replaced until the image is "moved).  The artifacts are present on both the google street photosphere and the Theta images.  It's as if the tiles are not readied prior to display (reminiscent of using Google Maps over a slow link).

Example: [Zoom test/Tile artifacts](http://tinyurl.com/j9ck2qr) <- from Theta
and: [Zoom test/Tile artifacts](http://tinyurl.com/z68k8k3)       <- from Google street view

Example: [Zoom issue only](http://tinyurl.com/hj6ehwa)  <- uses CDN (release 2.2)

**Comment by MmmDee** [comment]
This error showed up in the console during one of the artifacts...

`multiresFoyerYaw.html:1 [.CommandBufferContext.Offscreen-MainThread-02F9AE38]RENDER WARNING: texture bound to texture unit 0 is not renderable. It maybe non-power-of-2 and have incompatible texture filteri`ng.

**Comment by strarsis** [comment]
This may be related (or not), but the error message is idential:
https://github.com/mpetroff/pannellum/issues/178#issuecomment-229319241

**Comment by mpetroff** [comment]
I fixed the artifacts in 14682461e12b796842fd1580e8507c196821f44f. A recent change in texture loading had a race condition; depending on the timing, the artifacts either didn't happen, happened without a warning, or happened with a warning.

**Comment by mpetroff** [comment]
The maximum zoom for multires images is determined by a combination of the image size and the viewer size. When switching to full screen, the viewer size becomes bigger, so the maximum zoom allowed zoom is smaller; this prevents textures from being displayed too big, e.g. a 512px texture won't take up more than 512px on the screen.

The zoom restriction wasn't being checked when the viewer was being resized; it was only being checked when the zoom was changed, hence the jump. I fixed the bug by adding a check when the viewer is resized, e.g. full screen mode is activated.
